api/src/test/java/org/openmrs/validator/ObsValidatorTest.java

```diff
@@ -513,4 +516,36 @@ public void validate_shouldPassValidationIfValueTextIsLessThanTheMaximumLength()
 		Assert.assertFalse(errors.hasFieldErrors("obsDatetime"));
 		Assert.assertFalse(errors.hasFieldErrors("valueText"));
 	}
+
+	/**
+	 * @see ObsValidator#validate(Object, Errors)
+	 */
+	@Test
+	//@Verifies(value = "should add person with encounter patient for a new obs", method = "saveObs(Obs,String)")
+	public void saveObs_shouldUpdateObsPersonValueWithEncounterPatientForNewObs() {
+		ObsService os = Context.getObsService();
+		ConceptService cs = Context.getConceptService();
+		EncounterService es = Context.getEncounterService();
+		PersonService ps = Context.getPersonService();
+
+		Obs obs = new Obs();
+		obs.setConcept(cs.getConcept(3));
+		obs.setDateCreated(new Date());
+		obs.setCreator(Context.getAuthenticatedUser());
+		obs.setLocation(new Location(1));
+		obs.setObsDatetime(new Date());
+		obs.setValueText("Testing");
+		obs.setPerson(ps.getPerson(2)); // Setting this person (2) as different from that of encounter person (7)
+		obs.setEncounter(es.getEncounter(2));
+
+		Errors errors = new BindException(obs, "obs");
+		new ObsValidator().validate(obs, errors);
+
+		Assert.assertFalse(errors.hasFieldErrors("person"));
```

Comment:
assertFalse is already statically imported so `assertFalse(errors.hasFieldErrors("person")); `should be enough